pkg/types/validation/machinepools.go

```diff
@@ -38,7 +38,7 @@ func ValidateMachinePool(p *types.MachinePool, fldPath *field.Path, platform str
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("name"), p.Name, validMachinePoolNameValues))
 	}
 	if p.Replicas != nil {
-		if *p.Replicas <= 0 {
+		if *p.Replicas < 0 {
```

Comment:
I'm not sure why this is necessary of all a sudden.  It worked fine in v0.6.0.  Does not work in v0.7.0-master or master.

Reply:
Validation was added in v0.7.0.